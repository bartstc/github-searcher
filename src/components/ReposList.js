import React from 'react';
import RepoItem from './RepoItem';

const ReposList = ({ repos }) => (
  <ul className="list-group mt-2">
    {Array.isArray(repos) && repos.length === 0 && <h4 className="text-warning">No results</h4>}
    {Array.isArray(repos) && repos.map(repo => (
      <RepoItem key={repo.id} repo={repo} />
    ))}
  </ul>
);

export default ReposList;