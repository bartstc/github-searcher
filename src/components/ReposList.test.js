import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReposList from './ReposList';

describe('ReposList', () => {
  let props = { repos: {} };
  let list = mount(<ReposList {...props} />);

  it('renders properly', () => {
    expect(toJson(list)).toMatchSnapshot();
  });

  describe('list content after receive repos prop', () => {
    it('initializes the empty list of items without `No results` warning', () => {
      expect(list.props().repos).toEqual({});
    });

    it('renders `RepoItem` component when repos is not empty array', () => {
      props = {
        repos: [{
          owner: { login: 'test' },
          html_url: 'test',
          full_name: 'test',
          description: 'test',
          language: 'test'
        }]
      };
      list = shallow(<ReposList {...props} />);
      const ul = list.find('.list-group');

      expect(ul.find('Connect(RepoItem)').exists()).toBe(true);
    });

    it('renders warning when no repos found', () => {
      props = { repos: [] };
      list = shallow(<ReposList {...props} />);

      expect(list.find('.text-warning').text()).toEqual('No results');
    });
  })
});