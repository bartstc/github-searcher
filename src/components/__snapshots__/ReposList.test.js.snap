// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposList renders properly 1`] = `
<ReposList
  repos={Object {}}
>
  <ul
    className="list-group mt-2"
  />
</ReposList>
`;
